olive/cli/quantize.py

```diff
@@ -76,11 +84,92 @@ def _get_run_config(self, tempdir: str) -> Dict[str, Any]:
         update_dataset_options(self.args, config)
         update_shared_cache_options(config, self.args)
 
+        is_hfmodel = config["input_model"]["type"].lower() == "hfmodel"
+        if is_hfmodel and self.args.algorithm not in ["awq", "gptq", "rtn"]:
+            raise ValueError("Selected algorithm is not supported for HuggingFace models.")
+
+        if self.args.algorithm == "awq":
+            if not self.args.implementation:
+                self.args.implementation = "awq" if is_hfmodel else "nvmo"
+            if not self.args.precision:
+                self.args.precision = "int4"
+
+        elif self.args.algorithm == "gptq":
+            if not self.args.implementation:
+                self.args.implementation = "gptq" if is_hfmodel else "matmul4"
+            if not self.args.precision:
+                self.args.precision = "int16" if self.args.implementation == "quarot" else "int4"
+
+        elif self.args.algorithm == "rtn":
+            if not self.args.implementation:
+                self.args.implementation = "quarot" if is_hfmodel else "bnb4"
+            if not self.args.precision:
+                self.args.precision = "int16" if self.args.implementation == "quarot" else "nf4"
+
+        elif self.args.algorithm == "hqq":
+            if not self.args.implementation:
+                self.args.implementation = "matmul4"
+            if not self.args.precision:
+                self.args.precision = "int4"
+
+        elif self.args.algorithm == "static" or self.args.algorithm == "dynamic":
+            if not self.args.implementation:
+                self.args.implementation = "onnx_static"
+            if not self.args.precision:
+                self.args.precision = "int8"
+
+        if self.args.enable_qdq_encoding and (
+            self.args.algorithm in ["rtn", "gptq", "hqq"] or self.args.implementation == "matmul4"
+        ):
+            self.args.implementation = "mnb_to_qdq"
```

Comment:
this needs to be added to the passflow that already contains `matmul4`. it is not a replacement for it.